lightclient/src/background.rs

```diff
@@ -1,43 +1,47 @@
-// Copyright 2019-2023 Parity Technologies (UK) Ltd.
+// Copyright 2019-2024 Parity Technologies (UK) Ltd.
 // This file is dual-licensed as Apache-2.0 or GPL-3.0.
 // see LICENSE for license details.
 
-use futures::stream::StreamExt;
-use futures_util::future::{self, Either};
-use serde::Deserialize;
+use crate::rpc::RpcResponse;
+use crate::shared_client::SharedClient;
+use crate::{JsonRpcError, LightClientRpcError};
+use futures::{stream::StreamExt, FutureExt};
 use serde_json::value::RawValue;
 use smoldot_light::platform::PlatformRef;
 use std::{collections::HashMap, str::FromStr};
 use tokio::sync::{mpsc, oneshot};
+use tokio_stream::wrappers::UnboundedReceiverStream;
 
-use crate::client::AddedChain;
-
-use super::LightClientRpcError;
-use smoldot_light::ChainId;
-
-const LOG_TARGET: &str = "subxt-light-client-background";
+const LOG_TARGET: &str = "subxt-light-client-background-task";
 
 /// The response of an RPC method.
 pub type MethodResponse = Result<Box<RawValue>, LightClientRpcError>;
 
+/// The response back from an RPC subscription call; a subscription ID and channel that
+/// subscription notifications will be received on.
+pub type SubscriptionResponse = Result<
+    (
+        SubscriptionId,
+        mpsc::UnboundedReceiver<Result<Box<RawValue>, JsonRpcError>>,
+    ),
+    LightClientRpcError,
+>;
+
+/// Type of subscription IDs we can get back.
+pub type SubscriptionId = String;
```

Comment:
It was previously an unsigned int, but even if Smoldot _currently_ does that, I didn't think we should rely on it since it probably also expects us to treat the valeus as opaque.